.travis/update_checkstyle_thresholds

```diff
@@ -74,15 +78,44 @@ has_local_thresholds_changed() {
 }
 
 update_remote_thresholds() {
-  echo "Pushing '$GRADLE_PROPERTIES' changes to '$TRAVIS_REPO_SLUG:$TRAVIS_BRANCH'..."
   local -r old_hash=$1
-  local -r update_request="{ \
+  local -r branch_name="checkstyle_update_thresholds_after_build_${TRAVIS_BUILD_NUMBER}"
+
+  create_remote_branch "$branch_name"
+  update_remote_file "$branch_name" "$old_hash"
+  create_remote_pull_request "$branch_name"
+}
+
+create_remote_branch() {
+  local -r branch_name=$1
+
+  echo "Creating branch '$branch_name' in '$TRAVIS_REPO_SLUG' at SHA '$TRAVIS_COMMIT'..."
+  local -r create_ref_request="{ \
+    \"ref\": \"refs/heads/$branch_name\", \
+    \"sha\": \"$TRAVIS_COMMIT\"
```

Comment:
I'm curious why this is working without the backslash before the newline.

Reply:
Apparently, escaping the newlines within a quoted string is unnecessary for "line continuation."  This is perfectly valid bash:

```bash
$ a="123
> 456"
$ echo "$a"
123
456
```

So all I'm accomplishing by escaping the newlines is removing them from the actual JSON string sent in the request (except for the last one I didn't escape).  I'll clean up all these variable initializations.